pkg/provisioner/ironic/ironic.go

```diff
@@ -1212,6 +1245,23 @@ func (p *ironicProvisioner) Provision(hostConf provisioner.HostConfigData) (resu
 		return transientError(provisioner.NeedsRegistration)
 	}
 
+	if p.host.Spec.AutomatedCleaningMode == metal3v1alpha1.CleaningModeDisabled {
+		p.log.Info("automated cleaning is disabled for host",
+			"ID", ironicNode.UUID,
+			"automated_clean", ironicNode.AutomatedClean,
+		)
+		p.publisher("CleaningDisabled", "Automated cleaning disabled")
+	}
+
+	if ironicNode.AutomatedClean != nil {
```

Comment:
It's `nil` by default. Not sure what you meant here, but it seems like it will never be updated?

Reply:
@dtantsur This was added here to avoid NPE. Removing `if ironicNode.AutomatedClean != nil` will cause runtime panic. (example [snippet](https://paste.ubuntu.com/p/MKXGcgGDZK/)). Maybe `ironicNode.AutomatedClean` in gophercloud should be defaulted to `True` as you mentioned during the review on the PR in gophercloud, or I need to assign `True` here before my logic comes into play.

Reply:
> It's `nil` by default. Not sure what you meant here, but it seems like it will never be updated?

It seems so.

Reply:
I did some code refactoring to verify if that's the issue, and indeed we are not able to update it just because ironicNode.AutomatedClean is `nil`

Reply:
As discussed on the [slack](https://kubernetes.slack.com/archives/CHD49TLE7/p1615195502083900), I changed the code a bit to enforce always updating the Ironic node.